packages/cli/test/app-run.test.js

```diff
@@ -0,0 +1,57 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const path = require('path');
+const assert = require('yeoman-assert');
+const helpers = require('yeoman-test');
+
+const lerna = require('lerna');
+function lernaBootstrap() {
+  const cmd = new lerna.BootstrapCommand('', {
+    loglevel: 'silent',
+  });
+  return cmd.run();
+}
+
+const runShell = require('@loopback/build').runShell;
+
+describe('app-generator', function() {
+  const generator = path.join(__dirname, '../generators/app');
+  const rootDir = path.join(__dirname, '../../..');
+  const sandbox = path.join(__dirname, '../../_sandbox');
+  const cwd = process.cwd();
+  const props = {
+    name: 'myApp',
+    description: 'My app for LoopBack 4',
+    outdir: sandbox,
+  };
+
+  // WARNING: It takes a while to run `lerna bootstrap`
+  this.timeout(0);
+  before(() => {
+    return helpers
+      .run(generator)
+      .inDir(sandbox)
+      .withPrompts(props);
+  });
+
+  it('passes `npm test` for the generated project', async () => {
+    process.chdir(rootDir);
+    await lernaBootstrap();
+    process.chdir(sandbox);
+    return new Promise((resolve, reject) => {
+      runShell('npm', ['test', '--', '--allow-console-logs']).on(
```

Comment:
I think `npm test` might be overkill from #932. In the acceptance criteria, we just wanted to run `npm start` and make a request or two to make sure the generated app works. However, running `npm test` basically does both in one shot, even if it's slow.

Reply:
The expensive operation is `lerna bootstrap`. Running `npm start` is not going to help us:

1. `npm start` starts the app, but it's hard to verify it's functional.
2. We have to stop the app

`npm test` does what we need with automation.